### PR TITLE
[Qt] Add General Page to Settings

### PR DESCRIPTION
**Pull request by Ryanel**
Adds the "General" Page to the Settings Dialog.
screenshot from 2016-05-12 02-13-08

The only thing that doesn't work to my knowledge is the Language selector, but that's due to me not knowing much about translation.

Also, this time, I tried not to make so many mistakes :p

---

This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3836)

### CODE REVIEW

**Comment by rukai** [comment]
Now that PR's are being merged again, I'll pick up my translations PR again. https://github.com/dolphin-emu/dolphin/pull/3312
For now just keep the language option as a dummy.

**Comment by lioncash** [comment]
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[Source/Core/DolphinQt2/Config/SettingsWindow.cpp, line 7 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbawH418_YnURzM3cw:-KHbawH5TeDY31uGfm60:1478904044) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/SettingsWindow.cpp#L7)):*
> ```cpp
> #include "DolphinQt2/Settings.h"
> #include "DolphinQt2/Config/SettingsWindow.h"
> #include "DolphinQt2/Config/Pages/GeneralPage.h"
> ```

Newline after this header include.

---

*[Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp, line 6 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbbUORvjnKjrhuMdE4:-KHbbUOS5XwhNh90e1EX:1675598875) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp#L6)):*
> ```cpp
> #include <QLabel>
> #include <QString>
> #include <QVBoxLayout>
> ```

Missing an include for QHBoxLayout

---

*[Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp, line 33 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbb9kIghbnuiKStVoh:-KHbb9kIghbnuiKStVoi:-305512373) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp#L33)):*
> ```cpp
> 
> 	u32 selection = qRound(startup_params.m_EmulationSpeed * 10.0f);
> 	if (selection < (u32)m_checkbox_speed_limit->count())
> ```

static_cast

---

*[Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp, line 88 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbbH70i2lEQKgP_CkI:-KHbbH70i2lEQKgP_CkJ:188368419) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp#L88)):*
> ```cpp
> 		QString str;
> 		if (i == 100)
> 			str.sprintf("%i%% (Normal Speed)",i);
> ```

space after the comma

---

*[Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp, line 90 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbbFHwNPR-S2DN6apE:-KHbbFHwNPR-S2DN6apF:188368419) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/Pages/GeneralPage.cpp#L90)):*
> ```cpp
> 			str.sprintf("%i%% (Normal Speed)",i);
> 		else
> 			str.sprintf("%i%%",i);
> ```

space after the comma

---

*[Source/Core/DolphinQt2/Config/Pages/GeneralPage.h, line 32 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836#-KHbb66RgvszgqCKsdSB:-KHbb66RgvszgqCKsdSC:-1094895339) ([raw file](https://github.com/dolphin-emu/dolphin/blob/c79c70135dabaa6d0a42496c919a387e40d427df/Source/Core/DolphinQt2/Config/Pages/GeneralPage.h#L32)):*
> ```objc
> 	// Advanced Group
> 	QCheckBox* m_checkbox_force_ntsc;
> };
> ```

Missing newline

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3836)*

File: Source/Core/DolphinQt2/DolphinQt2.vcxproj [hunk 1]
```diff
@@ -111,6 +111,7 @@
     <ClCompile Include="$(QtMocOutPrefix)TableDelegate.cpp" />
     <ClCompile Include="$(QtMocOutPrefix)ToolBar.cpp" />
     <ClCompile Include="AboutDialog.cpp" />
+    <ClCompile Include="Config\Pages\GeneralPage.cpp" />
```
Comment (Parlane) [inline Source/Core/DolphinQt2/DolphinQt2.vcxproj]: Also need to add a QtMoc item above.

**Comment by RisingFog** [comment]
What's the status of this PR?

**Pull request closed without merging**